Update index.rst (#853)
flyteorg/flyte: rsts/index.rst

> .. image:: images/flyte_lockup_gradient_on_light.png

Reviewer: I think we can get rid of this logo and the above

```
Flyte
=====
```

Title and have the main page start with `Meet Flyte`

With the updated theming I'll push up after this the branding will be clearer:

![image](https://user-images.githubusercontent.com/2816689/112322242-1e320400-8c87-11eb-8e8a-8516ddcb432a.png)

Author: Yes! Let's plan this with the new layout.

Meet Flyte
==========

Flyte is an open-source, container-native, structured programming and distributed processing platform. It enables highly concurrent, scalable and maintainable workflows for machine learning and data processing.

> Created at Lyft, Flyte provides first class support for Python, and is built directly on Kubernetes for all the benefits containerization provides: portability, scalability, and reliability.

Reviewer 2: Wouldn't it be better if we include Java & Scala as well?

Author: Thank you @cosmicBboy for including these changes below.

> Created at Lyft, Flyte provides first class support for Python, and is built directly on Kubernetes for all the benefits containerization provides: portability, scalability, and reliability.

Reviewer: ```suggestion
Created at Lyft, Flyte provides first class support for Python, Java, and Scala, and is built directly on Kubernetes for all the benefits containerization provides: portability, scalability, and reliability.
```

> Flyte provides a single unit of execution (task) as a top-level concept. Multiple tasks arranged in a data producer-consumer order create the workflow, which is pure specification created in any language.

Reviewer 2: ...which is a pure specification...

Reviewer 2: I believe we can include a short code snippet depicting `task` and `workflow` -- just to visualize the theoretical explanation.

Author: Thank you @cosmicBboy for including them below.

Why Flyte?
==========

> Flyte's main purpose is to increase the development velocity for data processing and machine learning, enabling large-scale compute execution without the operational overhead. Teams can therefore focus on the business goals and not the infrastructure.

Reviewer 2: ...large-scale compute-intensive executions without...

Author: Done.

Reviewer 3: Maybe in this line add a little bit about built-in integrations with your favourite Saas technologies.

So something like,
Flyte is built to be a fabric that connects disparate compute technologies ranging from cloud hosted to open source and it provides custom implementations where appropriate

What makes Flyte different?
---------------------------

> * Fully Containerized

Reviewer 3: ```suggestion
* Container Native
```

> * Extensible Backend & SDK’s

Reviewer 2: ...SDKs...

Author: Done.

> * Ergonomic SDK’s in Python and Java/Scala

Reviewer 2: ...SDKs...

Author: Done.

> * Ergonomic SDK’s in Python and Java/Scala

Reviewer 3: ```suggestion
* Ergonomic SDK’s in Python, Java & Scala
```

> * Versioned & Auditable - all actions are recorded

Reviewer 3: Guaranteed reproducibility of pipelines via:
- Versioned data, code and models
- Automatically tracked executions
- Declarative pipelines

Author: done

> * Horizontally Scalable - execute a single task or a workflow, attach multiple schedules to a workflow

Reviewer 3: ```suggestion
* Matches your workflow - Start with one task, convert to a pipeline, attach multiple schedules or trigger using a programmatic API or on-demand
```

* Battle-tested - millions of pipelines executed per month
* Vertically-Integrated Compute - serverless experience

> * Deep understanding of data-lineage & provenance

Reviewer 3: Instead of this lets say memoize results across executions

Author: Done.

* Operation Visibility - cost, performance, etc.
* Cross-Cloud Portable Pipelines

> 

Reviewer 2: Other points that you could add:
* Memoization
* Dynamic Workflows

Author: Done.

> * Cross-Cloud Portable Pipelines

Reviewer 3: we need to add collaboration as an important item

**Collaborate across your organization**
Execute distributed data pipelines / workflows
Reuse tasks across projects, users and workflows
Clearly trace which workflow steps produced the results
Compare results of training workflows over time and across pipelines.
Share workflows and tasks across your teams

Author: Done.

> 

Reviewer 3: Subscribe to events for your pipelines and receive notifications on popular channels like Slack, pagerduty and email

Author: Done.

> At Lyft, Flyte has served production model training and data processing for over four years, becoming the de-facto platform for the Pricing, Locations, ETA and Mapping teams.

Reviewer 3: ```suggestion
At Lyft, Flyte has served production model training and data processing for over four years, becoming the de-facto platform for the Pricing, Locations, ETA, Mapping teams, Growth, Autonomous and other teams
```

> Whether you will be writing Flyte workflows, deploying the Flyte platform to your k8 cluster, or would like to extend and contribute to the architecture and design of Flyte, we have what you need.

Reviewer 2: I believe we'll have to talk about how we're going to simplify the user job. How about the following line? --

Be it seamlessly orchestrating your data processing or machine learning pipeline, caching the pre-computed artifacts, or inducing dynamism into your workflows, Flyte provides an off-the-shelf infrastructure to push your models to production in no time!

Author: Done.

Welcome to the documentation hub for Flyte!

> Beginners: see Getting Started

Reviewer 2: How about Including links here? Say,
* For Beginners (here you can provide a link)
* ...

Author: The plan is to have the words "Getting Started" with the embedded link.

> Beginners: see Getting Started

Reviewer 4: Shouldn't these be links to those sections?

Author: Yes there should be links. I did not include them now because they will change once we update the website. Do you think we should include them and then change them later?

Reviewer 4: yes, let's make them links... You should be able to use cross referencing here so you don't have to hardcode a link...
e.g. 
```suggestion
Beginners: see :ref:`getting-started-firstrun`
```

> Beginners: see :ref:`getting-started-firstrun`

Reviewer 3: ```suggestion
First time: see :ref:`getting-started-firstrun`
```

Author: Edited

> Intermediate Users: see :ref:`plugins`

Reviewer 3: This is not really intermediate
`Checkout available integrations `

Author: Edited

> Advanced Users: see :ref:`divedeep`

Reviewer 3: Dive deep into the architecture of Flyte

> Advanced USers: see Architecture and Concepts

Reviewer 4: ```suggestion
Advanced Users: see Architecture and Concepts
```
